graspologic/nominate/spectralVN.py

```diff
@@ -0,0 +1,263 @@
+
+    References
+    ----------
+    .. [1] Fishkind, D. E.; Lyzinski, V.; Pao, H.; Chen, L.; Priebe, C. E. Vertex nomination schemes for membership
+        prediction. Ann. Appl. Stat. 9 2015. https://projecteuclid.org/euclid.aoas/1446488749
+
+    .. [2] Jordan Yoder, Li Chen, Henry Pao, Eric Bridgeford, Keith Levin, Donniell E. Fishkind, Carey Priebe,
+        Vince Lyzinski, Vertex nomination: The canonical sampling and the extended spectral nomination schemes,
+        Computational Statistics & Data Analysis, Volume 145, 2020.
+        http://www.sciencedirect.com/science/article/pii/S0167947320300074
+
+    """
+
+    def __init__(
+        self,
+        embedding: np.ndarray = None,
+        embedder: Union[str, BaseSpectralEmbed] = "ASE",
+        persistent: bool = True,
+    ):
+        super().__init__(multigraph=False)
+        self.embedding = embedding
+        if self.embedding is None or not persistent:
+            if isinstance(embedder, BaseSpectralEmbed):
+                self.embedder = embedder
+            elif embedder == "ASE":
+                self.embedder = ase()
+            elif embedder == "LSE":
+                self.embedder = lse()
+            else:
+                raise TypeError
+        elif np.ndim(embedding) != 2:
+            raise IndexError("embedding must have dimension 2")
+        self.persistent = persistent
+        self.attribute_labels_ = None
+        self.unique_attributes_ = None
+        self.distance_matrix_ = None
+        self.neigh_inds = None
+
+    @staticmethod
+    def _make_2d(arr: np.ndarray) -> np.ndarray:
+        # ensures arr is two or less dimensions.
+        # if 1d, adds unique at each index on
+        # the second dimension.
+        if not np.issubdtype(arr.dtype, np.integer):
+            raise TypeError("Argument must be of type int")
+        arr = np.array(arr, dtype=np.int)
+        if np.ndim(arr) > 2 or (arr.ndim == 2 and arr.shape[1] > 2):
+            raise IndexError("Argument must have shape (n) or (n, 1) or (n, 2).")
+        elif np.ndim(arr) == 1 or arr.shape[1] == 1:
+            arr = arr.reshape(-1, 1)
+            arr = np.concatenate((arr, np.arange(arr.shape[0]).reshape(-1, 1)), axis=1)
+        return arr
+
+    def _embed(self, X: np.ndarray):
+        if not self.multigraph:
+            if not np.issubdtype(X.dtype, np.number):
+                raise TypeError("Adjacency matrix should have numeric type")
+            if np.ndim(X) != 2:
+                raise IndexError("Argument must have dim 2")
+            if X.shape[0] != X.shape[1]:
+                raise IndexError("Adjacency Matrix should be square.")
+        else:
+            raise NotImplementedError("Multigraph SVN not implemented")
+
+        # Embed graph if embedding not provided
+        if self.embedding is None:
+            if isinstance(self.embedder, BaseSpectralEmbed):
+                self.embedding = self.embedder.fit_transform(X)
+            else:
```

Comment:
If they provide an object to embedder that isn't of type BaseSpectralEmbed

Reply:
since we're using set_params here I say just take all type checking out of __init__ to avoid confusion, which is usually sklearn best practice anyway. in which case the type checking here needs to be more thorough. As do your error statements. For example, the error messages for invalid BaseSpectralEmbed vs for invalid string should be different and specific. It should be clear to users why the error is happening.